qemu/qmp.go

```diff
@@ -629,8 +629,9 @@ func (q *QMP) ExecuteBlockdevAdd(ctx context.Context, device, blockdevID string)
 // using the device_add command.  blockdevID should match the blockdevID passed
 // to a previous call to ExecuteBlockdevAdd.  devID is the id of the device to
 // add.  Both strings must be valid QMP identifiers.  driver is the name of the
-// driver,e.g., virtio-blk-pci, and bus is the name of the bus.  bus is optional.
-func (q *QMP) ExecuteDeviceAdd(ctx context.Context, blockdevID, devID, driver, bus string) error {
+// driver,e.g., virtio-blk-pci, bus is the name of the bus and addr is the address
+// where the device is plugged. bus and addr are optional.
+func (q *QMP) ExecuteDeviceAdd(ctx context.Context, blockdevID, devID, driver, bus string, addr string) error {
```

Comment:
You could just do

driver, bus, addr string

@mcastelino I wonder at what point we're going to need to provide a structure for the parameters to this function.  Are we likely to be adding more fields in the future?

Reply:
bootindex is another field that can be specified, but for now we don't need it